spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/config/ContextFunctionCatalogAutoConfiguration.java

```diff
@@ -164,10 +150,35 @@ private boolean isConverterEligible(Object messageConverter) {
 		if (messageConverterName.startsWith("org.springframework.cloud.")) {
 			return true;
 		}
-		else if (!messageConverterName.startsWith("org.springframework.")) {
-			return true;
+		return !messageConverterName.startsWith("org.springframework.");
+	}
+
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(name = "io.cloudevents.spring.messaging.CloudEventMessageConverter")
+	static class CloudEventsMessageConverterConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean
+		public CloudEventMessageConverter cloudEventMessageConverter() {
```

Comment:
ℹ️ putting these specific variants in a static class allows us to return the actual type from the bean method now as the class will not attempt to be loaded when not on classpath.